test-app/backend/src/shift/shift-api.ts

```diff
@@ -62,4 +66,65 @@ router.post('/new-shift', async (req, res) => {
   );
 });
 
+const errorCheckDates = (newShift: Partial<Shift>, origShift: Shift) => {
```

Comment:
Can you name this a bit differently to clarify the return value. The name implies that it will throw an error probably. Instead, you can name it like `isIntervalInvalid()` so it's easier to understand that it returns a boolean.

Also, it might be easier to just pass in 2 dates here instead of shifts (that way it's reusable). There's no actual need to pass in the shift itself since you never grab any properties other than the times.

I also think the logic here can be simplified. There are sorta 4 scenarios:
1. new start & new end
2. just new start
3. just new end
4. nothing new in terms of date

In each case, you just need to figure out what the new start date and end date would be of the shift. It doesn't really matter (as far as I can see at least) what the old vs new values are. Because of that, the checks for dates just need to be that they exist and that start < end.